### PR TITLE
[telemetrygen] Add extra documentation for key-value parameters

### LINKED ISSUE #21042: Key-value parameters are not recognized by telemetrygen
### Component(s)

cmd/telemetrygen

## Description
The `telemetrygen` tool doesn't recognize any key-value parameter provided. I tried with `otlp-header` and `otlp-attributes`.

## Steps to Reproduce
```sh
$ docker run ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest traces --otlp-header "key=value"
Error: invalid argument "key=value" for "--otlp-header" flag: value should be a string wrapped in double quotes
$ docker run ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest traces --otlp-header key="value"
Error: invalid argument "key=value" for "--otlp-header" flag: value should be a string wrapped in double quotes
$ docker run ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest traces --otlp-attributes key="value"
Error: invalid argument "key=value" for "--otlp-attributes" flag: value should be a string wrapped in double quotes
$ docker run ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest traces --otlp-attributes "key=value" 
Error: invalid argument "key=value" for "--otlp-attributes" flag: value should be a string wrapped in double quotes
```

Tested with bash and zsh.

### Collector version

latest

## Environment
```sh
$ cat /etc/fedora-release 
Fedora release 37 (Thirty Seven)
```

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iblancasa**
**Description:**
Resolves #21042.

### CODE REVIEW

File: cmd/telemetrygen/internal/common/config.go [hunk 1]
```diff
@@ -83,11 +83,13 @@ func (c *Config) CommonFlags(fs *pflag.FlagSet) {
 
 	// custom headers
 	c.Headers = make(map[string]string)
-	fs.Var(&c.Headers, "otlp-header", "Custom header to be passed along with each OTLP request. The value is expected in the format key=value."+
+	fs.Var(&c.Headers, "otlp-header", "Custom header to be passed along with each OTLP request. The value is expected in the format key=\"value\"."+
+		"Note you may need to scape the quotes when using the tool from a cli."+
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/common/config.go:87]:
```suggestion
		"Note you may need to escape the quotes when using the tool from a cli."+
```

File: cmd/telemetrygen/internal/common/config.go [hunk 2]
```diff
@@ -83,11 +83,13 @@ func (c *Config) CommonFlags(fs *pflag.FlagSet) {
...
 		"Flag may be repeated to set multiple headers (e.g -otlp-header key1=value1 -otlp-header key2=value2)")
 
 	// custom resource attributes
 	c.ResourceAttributes = make(map[string]string)
 	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=\"value\"."+
+		"Note you may need to scape the quotes when using the tool from a cli."+
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/common/config.go:93]:
```suggestion
		"Note you may need to escape the quotes when using the tool from a cli."+
```

**Review by mx-psi — APPROVED** [review]

**Comment by iblancasa** [comment]
Sorry for the typos.

**Comment by mx-psi** [comment]
No worries and thanks for the PR!

**Pull request merged**